src/Chapter3.hs

```diff
@@ -970,6 +1039,33 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Weekday = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Eq, Show)
+
+instance Enum Weekday where
+  succ Monday    = Tuesday
+  succ Tuesday   = Wednesday
+  succ Wednesday = Thursday
+  succ Thursday  = Friday
+  succ Friday    = Saturday
+  succ Saturday  = Sunday
+  succ Sunday    = Monday
```

Comment:
`Enum` class have some required methods that needs to be implemented. You would get runtime exception by trying using any other methods of the instance, which is not good.

Also, I am pretty sure that some of the function would be broken due to the infinitely dependant `succ`.

In this task particularly, we would want you to use `Enum` + `Bounded` instances of your data type. So if you would derive those two, all of your function will look much easier and would be maintainable better 🙂